worker/ocspStatus/ocspStatus.go

```diff
@@ -0,0 +1,167 @@
+	worker.RegisterWorker(workerName, worker.Info{Runner: runner, Description: workerDesc})
+}
+
+type Result struct {
+	Status    int       `json:"status"`
+	RevokedAt time.Time `json:"revoked_at,omitempty"`
+}
+
+type ocspStatus struct{}
+
+func (w ocspStatus) Run(in worker.Input, resChan chan worker.Result) {
+	res := worker.Result{
+		WorkerName: workerName,
+		Success:    false,
+	}
+
+	x509Cert, err := in.Certificate.ToX509()
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	issuerEncoded := in.CertificateChain.Certs[len(in.CertificateChain.Certs)-1] // last cert in chain
+	x509IssuerCertEncoded, err := base64.StdEncoding.DecodeString(issuerEncoded)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s decoding issuer for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+	x509Issuer, err := x509.ParseCertificate(x509IssuerCertEncoded)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s parsing issuer for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// grab OCSP response
+	opts := &ocsp.RequestOptions{
+		Hash: crypto.SHA256,
+	}
+	req, err := ocsp.CreateRequest(x509Cert, x509Issuer, opts)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s creating OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+	httpResponse, err := http.Post(x509Cert.OCSPServer[0], "application/ocsp-request", bytes.NewReader(req))
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s making OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	// parse response
+	output, err := ioutil.ReadAll(httpResponse.Body)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s reading OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+	resp, err := ocsp.ParseResponse(output, x509Issuer)
+	if err != nil {
+		resChan <- worker.Result{
+			Success:    false,
+			WorkerName: workerName,
+			Errors:     []string{fmt.Sprintf("%s parsing OCSP response for certificate %d", err.Error(), in.Certificate.ID)},
+			Result:     nil,
+		}
+		return
+	}
+
+	status := Result{
+		Status: resp.Status,
+	}
+	if resp.Status == ocsp.Revoked {
+		status.RevokedAt = resp.RevokedAt
+	}
```

Comment:
The `revoked_at` field is still stored here, but it looks to be detected by `.IsZero()`, which doesn't really make sense as a revocation time anyway.

The status field is always checked in the printer anyway. 

```
tls_observatory=# select * from analysis where worker_name='ocspStatus' and scan_id=5; 
 id | scan_id | worker_name | success |                       output                        
----+---------+-------------+---------+-----------------------------------------------------
 25 |       5 | ocspStatus  | t       | {"status": 0, "revoked_at": "0001-01-01T00:00:00Z"}
```